### ISSUE TITLE
Redshift resource needs a snapshot_account_owner property

### ISSUE THREAD

**Issue by csabatini** [hashicorp/terraform#11638]
The AWS provider Redshift resource has 'snapshot_identifier' and 'snapshot_cluster_identifier' properties for restoring snapshots.

However, in the AWS CLI there is another argument, 'owner-account', which is mandatory when the snapshot is owned by another account. When I try to restore an authorized snapshot from another account with Terraform, I get this error:

> An error occurred (ClusterSnapshotNotFound) when calling the RestoreFromClusterSnapshot operation: Snapshot not found for cluster.

I can restore successfully via AWS CLI, and  get the same error in the CLI if I do not pass the 'owner-account' argument. It would be nice to be able to do this in Terraform.

**Comment by csabatini** [comment]
More information

## Terraform Version
0.8.4

## Affected Resource(s)
aws_redshift_cluster - https://www.terraform.io/docs/providers/aws/r/redshift_cluster.html
## Debug Output
Request and response for redshift/RestoreFromClusterSnapshot

```
2017/02/06 16:24:46 [DEBUG] plugin: terraform: aws-provider (internal) 2017/02/06 16:24:46 [DEBUG] Matching ^aws:* with owner
2017/02/06 16:24:46 [DEBUG] plugin: terraform: aws-provider (internal) 2017/02/06 16:24:46 [DEBUG] Redshift Cluster restore cluster options: {
2017/02/06 16:24:46 [DEBUG] plugin: terraform:   AllowVersionUpgrade: true,
2017/02/06 16:24:46 [DEBUG] plugin: terraform:   AutomatedSnapshotRetentionPeriod: 0,
2017/02/06 16:24:46 [DEBUG] plugin: terraform:   ClusterIdentifier: "pre-prod-master-redshift",
2017/02/06 16:24:46 [DEBUG] plugin: terraform:   ClusterParameterGroupName: "pre-prod-master-redshift-parameter-group",
2017/02/06 16:24:46 [DEBUG] plugin: terraform:   ClusterSubnetGroupName: "pre-prod-master-redshift-subnet-group",
2017/02/06 16:24:46 [DEBUG] plugin: terraform:   IamRoles: ["arn:aws:iam::207399867264:role/pre-prod_master_redshift_role"],
2017/02/06 16:24:46 [DEBUG] plugin: terraform:   NodeType: "dc1.large",
2017/02/06 16:24:46 [DEBUG] plugin: terraform:   Port: 5439,
2017/02/06 16:24:46 [DEBUG] plugin: terraform:   PreferredMaintenanceWindow: "Sun:18:00-Sun:22:00",
2017/02/06 16:24:46 [DEBUG] plugin: terraform:   PubliclyAccessible: false,
2017/02/06 16:24:46 [DEBUG] plugin: terraform:   SnapshotClusterIdentifier: "qa-master-redshift",
2017/02/06 16:24:46 [DEBUG] plugin: terraform:   SnapshotIdentifier: "qa-master-redshift-snap-2017-02-01t21-58-47",
2017/02/06 16:24:46 [DEBUG] plugin: terraform:   VpcSecurityGroupIds: ["sg-a369b1df"]
2017/02/06 16:24:46 [DEBUG] plugin: terraform: }
2017/02/06 16:24:46 [DEBUG] plugin: terraform: aws-provider (internal) 2017/02/06 16:24:46 [DEBUG] [aws-sdk-go] DEBUG: Request redshift/RestoreFromClusterSnapshot Details:
2017/02/06 16:24:46 [DEBUG] plugin: terraform: ---[ REQUEST POST-SIGN ]-----------------------------
2017/02/06 16:24:46 [DEBUG] plugin: terraform: POST / HTTP/1.1
2017/02/06 16:24:46 [DEBUG] plugin: terraform: Host: redshift.us-east-1.amazonaws.com
2017/02/06 16:24:46 [DEBUG] plugin: terraform: User-Agent: APN/1.0 HashiCorp/1.0 Terraform/0.8.4

2017/02/06 16:24:46 [DEBUG] plugin: terraform: -----------------------------------------------------
2017/02/06 16:24:47 [DEBUG] plugin: terraform: aws-provider (internal) 2017/02/06 16:24:47 [DEBUG] [aws-sdk-go] DEBUG: Response redshift/RestoreFromClusterSnapshot Details:
2017/02/06 16:24:47 [DEBUG] plugin: terraform: ---[ RESPONSE ]--------------------------------------
2017/02/06 16:24:47 [DEBUG] plugin: terraform: HTTP/1.1 404 Not Found
2017/02/06 16:24:47 [DEBUG] plugin: terraform: Connection: close
2017/02/06 16:24:47 [DEBUG] plugin: terraform: Content-Length: 353
2017/02/06 16:24:47 [DEBUG] plugin: terraform: Content-Type: text/xml
2017/02/06 16:24:47 [DEBUG] plugin: terraform: Date: Mon, 06 Feb 2017 16:24:46 GMT
2017/02/06 16:24:47 [DEBUG] plugin: terraform: X-Amzn-Requestid: c64fe19f-ec88-11e6-8994-efecfa885266
2017/02/06 16:24:47 [DEBUG] plugin: terraform: 
2017/02/06 16:24:47 [DEBUG] plugin: terraform: <ErrorResponse xmlns="http://redshift.amazonaws.com/doc/2012-12-01/">
2017/02/06 16:24:47 [DEBUG] plugin: terraform:   <Error>
2017/02/06 16:24:47 [DEBUG] plugin: terraform:     <Type>Sender</Type>
2017/02/06 16:24:47 [DEBUG] plugin: terraform:     <Code>ClusterSnapshotNotFound</Code>
2017/02/06 16:24:47 [DEBUG] plugin: terraform:     <Message>Snapshot qa-master-redshift-snap-2017-02-01t21-58-47 not found for cluster qa-master-redshift.</Message>
2017/02/06 16:24:47 [DEBUG] plugin: terraform:   </Error>
2017/02/06 16:24:47 [DEBUG] plugin: terraform:   <RequestId>c64fe19f-ec88-11e6-8994-efecfa885266</RequestId>
2017/02/06 16:24:47 [DEBUG] plugin: terraform: </ErrorResponse>
2017/02/06 16:24:47 [DEBUG] plugin: terraform: 
2017/02/06 16:24:47 [DEBUG] plugin: terraform: -----------------------------------------------------
2017/02/06 16:24:47 [DEBUG] plugin: terraform: aws-provider (internal) 2017/02/06 16:24:47 [ERROR] Error Restoring Redshift Cluster from Snapshot: ClusterSnapshotNotFound: Snapshot qa-master-redshift-snap-2017-02-01t21-58-47 not found for cluster qa-master-redshift.
```

## Expected Behavior
Redshift cluster is successfully restored from snapshot owned by another account

## Actual Behavior

Error encountered
```
* aws_redshift_cluster.redshift_cluster: ClusterSnapshotNotFound: Snapshot qa-master-redshift-snap-2017-02-01t21-58-47 not found for cluster qa-master-redshift.
	status code: 404, request id: c64fe19f-ec88-11e6-8994-efecfa885266
```

**Comment by stack72** [comment]
Closed - this is going to be available in Terraform 0.8.8 via #12062
